src/views/Player.vue

```diff
@@ -103,11 +136,123 @@ export default defineComponent({
       });
     });
 
+    // scorecard
+
+    /**
+     * defines all the metrics to show in the scorecard here
+     */
+    const scorecardMetrics = computed(() => {
+      return [
+        {
+          name: "Correct",
+          icon: {
+            source: "correct",
+            class:
+              "text-green-500 h-7 bp-360:h-8 bp-500:h-10 lg:h-10 w-8 bp-360:w-8 bp-500:w-10 md:w-10 lg:w-10 place-self-center",
+          },
+          value: state.numCorrect,
+        },
+        {
+          name: "Wrong",
+          icon: {
+            source: "wrong",
+            class:
+              "text-red-500 h-8 bp-360:h-8 bp-500:h-10 lg:h-11 w-6 bp-360:w-6 bp-500:w-6 md:w-7 lg:w-8 place-self-center",
+          },
+          value: state.numWrong,
+        },
+      ];
+    });
+
+    /**
+     * progress value (0-100) to be passed to the Scorecard component
+     */
+    const scorecardProgress = computed(() => {
+      const totalAttempted = state.numCorrect + state.numWrong;
+      if (totalAttempted == 0) return null;
+      return (state.numCorrect / totalAttempted) * 100;
+    });
+
+    /**
+     * number of questions that have been answered
+     */
+    const numQuestionsAnswered = computed(() => {
+      return state.numCorrect + state.numWrong;
+    });
+
+    const numSurveyQuestions = computed(() => {
+      let count = 0;
+      state.questions.forEach((itemDetail) => {
+        if (itemDetail.survey) count += 1;
+      });
+      // for (itemDetail of state.questions) {
+      //   if (itemDetail.survey) count += 1;
+      // }
+      return count;
+    });
+
+    const areAllQuestionsSurvey = computed(() => {
+      return numSurveyQuestions.value == state.questions.length;
+    });
+
+    function calculateScorecardMetrics() {
+      let index = 0;
+      state.questions.forEach((itemDetail) => {
+        updateNumCorrectWrongSkipped(itemDetail, state.responses[index].answer);
+        index += 1;
+      });
+    }
+
+    function updateNumCorrectWrongSkipped(itemDetail: any, userAnswer: any) {
+      if (itemDetail.survey) {
+        return;
+      }
+      if (itemDetail.type == "mcq" && !isNaN(userAnswer)) {
+        const correctAnswer = itemDetail.correct_answer;
+        isEqual(userAnswer, correctAnswer)
+          ? (state.numCorrect += 1)
+          : (state.numWrong += 1);
```

Comment:
this is not valid - earlier mcq was integer and that is why we were applying this NaN check - now the check for both checkbox and mcq should be the same because for both of them the answer is a list of option indices

Reply:
okayy